### PR TITLE
Fix train.py

### PR DESCRIPTION
**Pull request by Sudhendra**
The modified methods for compute_metrics and calculate_metric_with_sklearn fixes the Value Error: inhomogenous shape.

1. The "calculate_metric_with_sklearn" is modified to handle logits with ndim=3 (which was the case several times) and exclude padding tokens (assuming -100 is the padding token ID). 

2. The "compute_metrics" is modified to unpack logits if it's a tuple (which seemed to be the case in several error generating step).

These modifications completely resolve the problems with training and it just works. 

I hope my contribution to this codebase helps future researchers.

### CODE REVIEW

**Comment by Zhihan1996** [comment]
Thank you very much! This is very helpful and I have merged it.

**Comment by Sudhendra** [comment]
Your welcome :). Happy to contribute.
